### DISCUSSION TITLE
using learncard with HEDERA DID (L1 public ledger)

### DISCUSSION THREAD

**Discussion by purbid** [learningeconomy/LearnCard#214] [Help]
Hi, 
I was using learncard with my L1 blockchain/public ledger hedera (https://hedera.com/), my understanding was that learncard is blockchain agnostic, so it can use any blockchain. But when I use a hedera did to issue VC and VP, I get a method not specified error. Turns out in https://github.com/learningeconomy/LearnCard/blob/1e4133fd5/packages/learn-card-core/src/wallet/plugins/didkit/types.ts#L4 the specified DID keys belong to a hardcoded list of specific blockchains. Is there a way to change this to add our specific DID method?

**Comment by Custard7** [comment]
Hey @purbid,

Great question! Yes, there is definitely a way. While LearnCard SDK supports a number of chains & DID methods out of the box, the true agnosticism comes from our Plugin system.

In your case, I would recommend making a plugin for your chain!

There are a few approaches. Essentially you will want to interact with the ID Control Plane:

https://docs.learncard.com/learn-card-sdk/learncard-core/control-planes/id

Here are the docs for writing plugins:

https://docs.learncard.com/learn-card-sdk/learncard-core/plugins

Let us know if you have any questions!
